tests/unit/s2n_async_pkey_test.c

```diff
@@ -63,6 +63,51 @@ static int async_handler_wipe_connection_and_apply(struct s2n_connection *conn)
     return S2N_FAILURE;
 }
 
+static int async_handler_sign_with_different_pkey_and_apply(struct s2n_connection *conn) {
+
+    /* Check that we have pkey_op */
+    EXPECT_NOT_NULL(pkey_op);
+
+    /* Retrieve cipher suite for the server connection */
+    struct s2n_cipher_suite *cipher_suite = conn->secure.cipher_suite;
+    POSIX_ENSURE_REF(cipher_suite);
+
+    /* Test only sign operation */
+    if (cipher_suite->key_exchange_alg != NULL && cipher_suite->key_exchange_alg->is_ephemeral) {
+        /* Extract pkey */
```

Comment:
Why does this mean it's a signing operation? Maybe wait for https://github.com/aws/s2n-tls/pull/2779 to be merged and use _get_op_type.

Can you verify / prove that this if statement is true at least once?

Reply:
For this @xonatius helped me gather needed info. 
"if you check `expected_cipher->key_exchange_alg->is_ephemeral` flag, the true will indicate that we'll run a sign operation and false that we'll use a decrypt"

So i did check by expecting a success from a sign call in the test. As it failed at the `s2n_async_pkey_op_apply`, it confirmed that we are going into true condition at least once.

Reply:
Since this is a unit test, if you don't want to wait for the actual API to be merged, why not check the op's type directly? Like pkey_op->type?

Reply:
Done